pulp_python/app/models.py

```diff
@@ -92,9 +111,128 @@ class PythonImporter(Importer):
+            inventory.add(content.filename)
+        return inventory
+
+    def _fetch_remote(self):
+        remote = []
+
+        metadata_urls = [urljoin(self.feed_url, 'pypi/%s/json' % project)
+                         for project in self.project_names]
+
+
+        for metadata_url in metadata_urls:
+            parsed_url = urlparse(metadata_url)
+
+            download = self.get_futures_downloader(metadata_url, os.path.basename(parsed_url.path))
+            #download_result = download.fetch()
+            download()
+
+            metadata = json.load(open(download.writer.path))
+            for version, packages in metadata['releases'].items():
+                for package in packages:
+                    remote.append(self._parse_metadata(metadata['info'], version, package))
+
+        return remote
+
+    def _find_delta(self, inventory, remote, mirror=True):
+        remote_keys = set([content['filename'] for content in remote])
+
+        additions_set = remote_keys - inventory
+        additions = [content for content in remote if content['filename'] in additions_set]
+        if mirror:
+            removals_set = inventory - remote_keys
+            removals = [content for content in remote if content['filename'] in removals_set]
+        else:
+            removals = []
+
+        return Delta(additions=additions, removals=removals)
+
+    def _parse_metadata(cls, project, version, distribution):
+
+        package = {}
+
+        package['filename'] = distribution['filename']
+        package['packagetype'] = distribution['packagetype']
+        package['name'] = project['name']
+        package['version'] = version
+        package['metadata_version'] = '3.0'
+        package['summary'] = project.get('summary')
+        package['description'] = project.get('description')
+        package['keywords'] = project.get('keywords')
+        package['home_page'] = project.get('home_page')
+        package['download_url'] = project.get('download_url')
+        package['author'] = project.get('author')
+        package['author_email'] = project.get('author_email')
+        package['maintainer'] = project.get('maintainer')
+        package['maintainer_email'] = project.get('maintainer_email')
+        package['license'] = project.get('license')
+        package['requires_python'] = project.get('requires_python')
+        package['project_url'] = project.get('project_url')
+        package['platform'] = project.get('platform')
+        package['supported_platform'] = project.get('supported_platform')
+        package['requires_dist'] = project.get('requires_dist', [])
+        package['provides_dist'] = project.get('provides_dist', [])
+        package['obsoletes_dist'] = project.get('obsoletes_dist', [])
+        package['requires_external'] = project.get('requires_external', [])
+        package['url'] = distribution['url']
+        package['digest'] = distribution['digests'].get('sha256')
```

Comment:
AFAIK, each distribution only has a `md5_digest` unless we are syncing from another Pulp

Reply:
ah, so this is a difference between warehouse and legacy: Warehouse includes both the md5 and the sha256 by default

Reply:
you know what, I think you are right, `md5_digest` is also present separately from the `digest` field from warehouse. If we use `md5_digest` it'll be backwards compatible to legacy. I'm convinced, will update.